### PR TITLE
ACM-6929 'changes not be saved' should be thrown out

### PR DESCRIPTION
**Pull request by jeswanke**

### CODE REVIEW

File: frontend/src/components/AcmDataForm.tsx [hunk 1]
```diff
@@ -134,6 +138,29 @@ export function AcmDataFormPage(props: AcmDataFormProps): JSX.Element {
...
+  if (!dataRef.current) {
+    dataRef.current = resources
+  } else {
+    dirtyRef.current = !isEqual(dataRef.current, resources)
+  }
+  useEffect(() => {
+    const preventUnload = (event: BeforeUnloadEvent) => {
+      if (dirtyRef.current) {
+        // NOTE: This message isn't used in modern browsers, but is required
+        const message = 'Sure you want to leave?'
```
Comment (KevinFCormier) [inline frontend/src/components/AcmDataForm.tsx:153]: Might as well use `t('changes.maybe.lost')` here just in case?

Comment (jeswanke) [reply frontend/src/components/AcmDataForm.tsx:153]: fixed the nit

**Comment by jeswanke** [comment]
@KevinFCormier -- combined ACM 6929 with ACM 3089
also simplified code by using an npm version of before unload:

react-beforeunload

**Comment by jeswanke** [comment]
done did and done

**Pull request merged**